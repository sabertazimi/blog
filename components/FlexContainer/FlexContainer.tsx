import { cx } from '@components/utils'
import type { CSSProperties, HTMLProps, ReactNode } from 'react'

interface Props extends HTMLProps<HTMLDivElement> {
  children: ReactNode
  className?: string
  style?: CSSProperties
}

function FlexContainer({
  children,
  className,
  ...props
}: Props): JSX.Element {
  return (
    <div
      className={cx(
        'flex-container p-auto container relative mx-auto my-0 h-full',
        className,
      )}
      {...props}
    >
      {children}
    </div>
  )
}

export default FlexContainer
