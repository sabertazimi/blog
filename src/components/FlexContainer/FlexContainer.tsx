import type { CSSProperties, HTMLProps, ReactNode } from 'react';
import React from 'react';

interface Props extends HTMLProps<HTMLDivElement> {
  className?: string;
  style?: CSSProperties;
  children: ReactNode;
}

const FlexContainer = ({
  className = '',
  style = {},
  children,
  ...props
}: Props): JSX.Element => (
  <div
    className={`flex-container container relative h-full mx-auto my-0 p-auto ${className}`}
    style={{ ...style }}
    {...props}
  >
    {children}
  </div>
);

export default FlexContainer;
