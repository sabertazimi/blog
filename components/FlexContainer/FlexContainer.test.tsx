import { render, screen } from '@utils'
import { axe } from 'jest-axe'
import FlexContainer from './FlexContainer'

describe('flexContainer', () => {
  it('should render correctly (snapshot)', () => {
    const { container } = render(
      <FlexContainer role="main">
        <h1>FlexContainer</h1>
      </FlexContainer>,
    )

    expect(container).toMatchSnapshot()
  })

  it('should render children correctly', () => {
    render(
      <FlexContainer role="main">
        <h1>FlexContainer</h1>
      </FlexContainer>,
    )

    expect(screen.getByRole('main')).toContainElement(
      screen.getByText('FlexContainer'),
    )
  })

  it('should render accessibility guidelines (AXE)', async () => {
    const { container } = render(
      <FlexContainer role="main">
        <h1>FlexContainer</h1>
      </FlexContainer>,
    )

    const a11y = await axe(container)

    expect(a11y).toHaveNoViolations()
  })
})
