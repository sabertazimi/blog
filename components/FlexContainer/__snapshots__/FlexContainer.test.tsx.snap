// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flexContainer should render correctly (snapshot) 1`] = `
<div>
  <div
    class="flex-container p-auto container relative mx-auto my-0 h-full"
    role="main"
  >
    <h1>
      FlexContainer
    </h1>
  </div>
</div>
`;
